tangos/input_handlers/pynbody.py

```diff
@@ -433,22 +434,130 @@ def _is_able_to_load(self, filepath):
+        "VXc", "VYc", "VZc",
+    )
+
+    def available_object_property_names_for_timestep(self, ts_extension, object_typetag):
+        h = self._construct_halo_cat(ts_extension, object_typetag)
+
+        return (
+            [
+                key for key in h[1].properties.keys()
+                if key not in self._excluded_precalculated_properties
+            ] + list(self._included_additional_properties)
+        )
 
 
+    def _get_map_child_subhalos(self, ts_extension):
+        h = self._construct_halo_cat(ts_extension, 'halo')
+        halo_children = defaultdict(list)
+        for halo in h:
+            iparent = halo.properties["hostHalo"]
+            if iparent == 0:
+                continue
+
+            halo_children[iparent].append(halo.properties["halo_id"])
+        return halo_children
+
+    def iterate_object_properties_for_timestep(self, ts_extension, object_typetag, property_names):
+        h = self._construct_halo_cat(ts_extension, object_typetag)
+
+        h.physical_units()
+
+        if "child" in property_names:
+            map_child_parent = self._get_map_child_subhalos(ts_extension)
+
+        for halo in h:
+            # Since PR88, Tangos has a finder_offset mapping the index of the halo in the pynbody catalog,
+            # and a finder_id which is the intrinsic id of the halo.
+            # In general, these can be different, in particular when using specific AHF options or running with MPI.
+
+            # Since these changes, the iteration loop and the property importer expect us to yield
+            # first the finder offset (halo_id in our case)
+            # and second the finder_id (ID in our case, this will actually be skipped by the property importer anyway).
+            all_data = [halo.properties["halo_id"], halo.properties["ID"]]
+
+            halo_props = halo.properties
+            for k in property_names:
+                if k == "parent":
+                    data = proxy_object.IncompleteProxyObjectFromFinderId(
+                        halo_props['halo_id'],
+                        'halo'
+                    )
+                elif k == "child":
+                    data = [
+                        proxy_object.IncompleteProxyObjectFromFinderId(ichild, 'halo')
+                        for ichild in map_child_parent[halo_i]
+                    ]
+                elif k == "shrink_center":
+                    data = np.array([halo_props[k] for k in ("Xc", "Yc", "Zc")])
+                elif k == "bulk_velocity":
+                    data = np.array([halo_props[k] for k in ("VXc", "VYc", "VZc")])
+                else:
+                    data = halo_props[k]
+                all_data.append(data)
+
+            yield all_data
+
+
+class GadgetAHFInputHandler(AHFInputHandler):
+    patterns = ["snapshot_???"]
+    auxiliary_file_patterns = ["*.AHF_particlesSTARDUST"]
+
 
+class RAMSESAHFInputHandler(RamsesCatalogueMixin, AHFInputHandler):
```

Comment:
Is it correct that RAMSES is all-caps like this? Either way, naming should definitely be made consistent between the mixin and this class.

Reply:
Technically, Ramses is an acronym (although a bad one), so could be spelled all caps.

However, all previous handlers and classes spelt it `Ramses`, and we might break existing databases with e.g. `RamsesHOP` if we change the spelling of their class name. So I think the easiest is to update the new handler to match the existing one (14cd3f1)